charmcraft/application/main.py

```diff
@@ -135,6 +157,67 @@ def _get_dispatcher(self) -> craft_application.application._Dispatcher:  # type:
 
         return dispatcher
 
+    def run(self) -> int:  # (too many branches due to error handling)
+        """Bootstrap and run the application."""
+        dispatcher = self._get_dispatcher()
+        craft_cli.emit.trace("Preparing application...")
+
+        return_code = 1  # General error
+        try:
+            command = cast(
+                CharmcraftCommand,
+                dispatcher.load_command(
+                    {"app": self.app, "services": self.services, "global_args": self._global_args}
```

Comment:
another candidate for customization for 1.1
is this the only change in this run() method?

Reply:
It is, and definitely.

Reply:
https://github.com/canonical/craft-application/issues/125